Cleanup projects for Swift 4

dmiedema opened the pull request:
Remove use of `String.characters` and use `Substring` where necessary to
match original behavior.
Accept Xcode suggested settings for all included xcodeprojs including
submodules for EL*.
Convert `open` to `public` as per warning cleanup as well as take
migrator actions (which introduced `!`).

Tests were not run as part of this migration.

dmiedema committed:
Cleanup projects for Swift 4

Remove use of `String.characters` and use `Substring` where necessary to
match original behavior.
Accept Xcode suggested settings for all included xcodeprojs including
submodules for EL*.
Convert `open` to `public` as per warning cleanup as well as take
migrator actions (which introduced `!`).

Tests were not run as part of this migration.

dmiedema committed:
Further warning/error cleanup. Rename Codable

This was a run of the Swift 4.2 migrator on the `ModuloKit` target.
Resulted in force unwrap cleanup (inserted from the initial `modulo`
target Swift 4.2 migration) as well as renaming `Codable` and `
Decodable` to `ELCodable` and `ELDecodable` in addition to associated
errors.

Since Apple took over/stole the `Codable`/`Decodable` names, `ELCodable`
now sacrafices the good names for namespaced ones.

bsneed commented on Modules/ELCodable/ELCodable/Decimal.swift:
```diff
@@ -66,42 +66,42 @@ extension Decimal /*: FloatingPointType*/ { // It complains about _BitsType miss
     }
     /// `true` iff `self` is negative.
     public var isSignMinus: Bool {
-        return ((value as Double).sign == .minus)
```
This entire file should move to use the Swift version of Decimal, thus becoming ...

```
extension Decimal {
    public var value: NSDecimalNumber {
        get {
            return (self as NSDecimalNumber)
        }
    }
}
```

and that's it.

bsneed reviewed (changes requested):
Also, tests need to be run.

dmiedema committed:
Update for feedback

Ensure `ModuloKitTests` all pass as well as run official Swift 4.2 migration on
all targets as well as update to use official `Decimal`.
